synapse/app/_base.py

```diff
@@ -208,12 +209,28 @@ def quit_with_error(error_string: str) -> NoReturn:
     sys.exit(1)
 
 
+# via https://peps.python.org/pep-3134/#enhanced-reporting
+def format_exception_chain(exc: BaseException) -> str:
+    if exc.__cause__:
+        result = format_exception_chain(exc.__cause__)
+        result += "\nThe above exception was the direct cause..."
+    elif exc.__context__:
+        result = format_exception_chain(exc.__context__)
+        result += "\nDuring handling of the above exception, ..."
+
+    return "".join(traceback.format_exception(exc))
```

Comment:
Shouldn't we be using `result` here somehow?

Reply:
Good shout! Completely overlooked this and we can completely simplify in favor of just `traceback.format_exception(exc)` which looks to do all of this logic for us anyway. Wish some unused variable lint would have yelled at me about this.